tests/bwc/test_rolling_upgrade.py

```diff
@@ -132,6 +135,19 @@ def _test_rolling_upgrade(self, path, nodes):
                 self.assertEqual(res[1][0], 'no match title')
                 self.assertEqual(res[1][1], {'name': 'matchMe name'})
 
+                # Dynamically added empty arrays and ignored object sub-columns are indexed with special prefix starting from 5.5
+                # Ensure that reading such columns work across all versions.
+                # Related to https://github.com/crate/crate/commit/278d45f176e7d1d3215118255cd69afd2d3786ee
+                c.execute('''
+                    SELECT author, o
```

Comment:
I think we should select the concrete columns to ensure they are correctly resolved.
```suggestion
                    SELECT author['dyn_empty_array'], o['dyn_ignored_subcol']
```

Reply:
I tried this in my first draft. These columns are not added to the schema and statement will fail.

Reply:
o['dyn_ignored_subcol'] actually works, only author['dyn_empty_array']is problematic. Will apply partially

Reply:
Ah right, empty arrays aren't selectable as they aren't added to the schema nor inside an ignored object. But selecting a subcolumn of an ignored object works.

Reply:
Pushed an update

Reply:
I will double check assertion on selecting ignored column sub-col. I tried both 
  `self.assertEqual(res[0][1], '1')` and  `self.assertEqual(res[0][1], 1)`
and it fails in both cases. 

Probably type resolution is not same across different versions, will have a closer look.

Reply:
```
FAIL: test_rolling_upgrade (test_rolling_upgrade.RollingUpgradeTest) [4.4.x -> 4.5.x]
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/var/lib/jenkins/workspace/CrateDB/qa/crate_qa_on_pr/tests/bwc/test_rolling_upgrade.py", line 34, in test_rolling_upgrade
    self._test_rolling_upgrade(path, nodes=3)
  File "/var/lib/jenkins/workspace/CrateDB/qa/crate_qa_on_pr/tests/bwc/test_rolling_upgrade.py", line 149, in _test_rolling_upgrade
    self.assertEqual(res[0][1], 1)
AssertionError: '1' != 1

```

Reply:
And it was hitting only 4.5 + path because we added https://github.com/crate/crate/commit/96988544c86 in 4.5.0

Reply:
I see. Can you then just change the value for the ignored sub-column to text to avoid any cast? this casting issue isn't related to the test at all, we just want to ensure we can select an ignored sub-column.